File: src/test/java/com/gitlab/daring/image/util/CommonUtilsTest.java
package com.gitlab.daring.image.util;

import org.junit.Test;

import java.util.function.Consumer;

import static com.gitlab.daring.image.util.CommonUtils.tryRun;
import static org.junit.Assert.assertEquals;

public class CommonUtilsTest {

	@Test
	public void testTryRun() {
		Exception[] errRef = new Exception[1];
		Consumer<Exception> errFunc = e -> errRef[0] = e;
		Exception e1 = new Exception("e1");
		tryRun(() -> { throw e1; }, errFunc);
		assertEquals(e1, errRef[0]);
	}

}

File: src/test/kotlin/com/gitlab/daring/image/util/CommonUtilsTest.kt
package com.gitlab.daring.image.util

import com.gitlab.daring.image.util.CommonUtils.closeQuietly
import org.junit.Assert.assertEquals
import org.junit.Test

class CommonUtilsTest {

    @Test
    fun testCloseQuietly() {
        closeQuietly(null)
        closeQuietly(AutoCloseable { throw Exception("err1") })
        val closed = booleanArrayOf(false)
        closeQuietly(AutoCloseable { closed[0] = true })
        assertEquals(true, closed[0])
    }

}